### PR TITLE
Twitter: Preserve commit message whitespace

### LINKED ISSUE #959: Twitter service should preserve double-newlines in commit messages
E.g. in https://twitter.com/streamsstandard/status/526780118052524032 there should be two newlines after .pipeTo(), like there is in the original commit message.

### PR DESCRIPTION
**Pull request by parshap**
When tweeting commit messages, preserve the original commit message whitespace and tweet it as-is.

Previously, `String#split(" ")` was used to process words, [which splits on *any* whitespace and with runs of contiguous whitespace characters ignored][1]. When the split array was turned back into a string with `Array#join(" ")`, original whitespace characters were lost and replaced by a single space character.

Now, `String#gsub` and a regular expression is used to process words, which will preserve the original whitespace.

Fixes #959.

/cc @domenic @kdaigle

[1]: http://ruby-doc.org/core-2.2.0/String.html#method-i-split

### CODE REVIEW

**Comment by kdaigle** [comment]
This looks reasonable to me!

@ptoomey3 can you take a look at this for a :+1:? A regex changed but I'm pretty confident, since it is only changing how we escape at-mentions, that it is good to go.

File: lib/services/twitter.rb [hunk 1]
```diff
@@ -42,9 +42,9 @@ def receive_push
         url = commit['url']
         # Strip out leading @s so that github @ mentions don't become twitter @ mentions
         # since there's zero reason to believe IDs on one side match IDs on the other
-        message = commit['message'].split(' ').map do |word|
-          (word.length > 1 && word[0] == '@') ? "@\u200b#{word[1..word.length]}" : word
-        end.join(' ')
+        message = commit['message'].gsub(/\B@[^ ]+/) do |word|
```
Comment (ptoomey3) [inline lib/services/twitter.rb]: Total nit, but would `[[:graph:]]` be more accurate here? For example:

``` ruby
"foob @\tbar".gsub(/\B@[^ ]+/) { |word| puts word; "@X#{word[1..word.length]}" }
@	bar
=> "foob @X\tbar"
```

vs

``` ruby
"foob @\tbar".gsub(/\B@[[:graph:]]+/) { |word| puts word; "@X#{word[1..word.length]}" }
=> "foob @\tbar"
```

Comment (parshap) [inline lib/services/twitter.rb]: You're right, `[^ ]` is definitely too specific. I looked at what Twitter themselves look for when matching mentions, and it looks like the `@` [must be followed by `[a-zA-Z0-9_]{1,20}`][tt].

I changed the regex to use `[^[:word:]]` and added tests for this behavior.

[tt]: https://github.com/twitter/twitter-text/blob/master/rb/lib/twitter-text/regex.rb#L122

**Comment by parshap** [comment]
I added a commit to address @ptoomey3's comment.

I also discovered that twitter matches `＠` (U+FF20 FULLWIDTH COMMERCIAL AT) in addition to a normal ASCII `@`, so I added that too.

**Comment by kdaigle** [comment]
Thanks @parshap for getting this fixed up! I'll get this deployed in the next day or two. :cake:

**Comment by domenic** [comment]
<3 <3 <3 @parshap

**Pull request merged**